### ISSUE TITLE
Style issue: factory name incorrectly capitalized

### ISSUE THREAD

**Issue by draytond** [esvit/ng-table#504]
This line violates javascript naming conventions, causing it to break builds that depend on JSCS or JSHINT:

$scope.tableParams = new TableParams(parameters, settings);

So when we create a new object, causes our build to fail with a JSCS error, saying that constructor functions are supposed to be capitalized.  This should be an easy one to fix, and will help loads of people use this with JSCS or JSHINT.

**Comment by tb** [comment]
or use ["newcap": false](https://jslinterrors.com/a-constructor-name-should-start-with-an-uppercase-letter]) ... its quire common on [github](https://github.com/search?q=newcap&type=Code&utf8=%E2%9C%93)

**Comment by draytond** [comment]
It is just a style convention, however our project style conventions do require caps on our constructors.  Others might not have the same rule, but for teams like ours, giving it a capital letter would make it easier to use in our projects.  Might improve adoption.

**Comment by tb** [comment]
1. You can rename injected services https://docs.angularjs.org/guide/di
2. You can create factory that inherits from ngTableParams, thats something what I do to add extra code like loading data, custom filters etc

```coffee
angular.module('app')
  .factory('MyNgTableParams', ['ngTableParams', (ngTableParams) ->
    class MyNgTableParams extends ngTableParams
  ])
```

**Comment by draytond** [comment]
Indeed, that's a fine workaround.  We found our own work around as well.  I only make the recommendation to make adoption for teams like mine more seamless, so such workarounds are not necessary.  It's not a bug per se, more just something I think will improve ease of integration, and bring it more in-line with widely held conventions.
